posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -110,58 +117,59 @@ async def insert_into_s3_activity(inputs: S3InsertInputs):
             interval_end=inputs.data_interval_end,
         )
 
-        with tempfile.NamedTemporaryFile() as local_results_file:
-            while True:
+        # Loop through the results using the iterator, write them to a local
+        # file, and upload the file to S3 when it reaches 50MB in size.
+        while True:
+            with create_temporary_file(compression=inputs.compression) as (reader, writer):
                 try:
-                    result = await results_iterator.__anext__()
-                except StopAsyncIteration:
-                    break
-
-                if not result:
-                    break
+                    while True:
+                        result = await anext(results_iterator)
```

Comment:
Why not `async for result in results_iterator`?

Reply:
It's semantically equivalent to what we are doing here, but with less nesting/lines: https://docs.python.org/3/reference/compound_stmts.html#async-for

Reply:
I can't quite get the control flow I want using `async for`. I'll leave as is for now, but narrow the exception handling type to `ChunkSizeLimitReached`